src/resmom/linux/mom_mach.c

```diff
@@ -8059,7 +8059,7 @@ pidcache_reset(pidcachetype_t *pidcache)
 		fclose(fd);
 	}
 
-	if (ncantstat)
+	if (!nprocs && ncantstat)
```

Comment:
No. It should return 0. Its not an error case not finding any job processes. mom will not have any processes to monitor and function should return successfully.